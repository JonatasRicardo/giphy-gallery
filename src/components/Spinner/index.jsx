
import React from 'react';
import { useTheme } from "react-jss";
import useStyles from "./spinner.styles";

const Spinner = () => {
    const theme = useTheme();
    const classes = useStyles({ theme });

    return (
    <div className={classes.spinner}>
      <div className={classes.doubleBounce} />
      <div className={[classes.doubleBounce, classes.doubleBounce2]}  />
    </div>
    );
}

export default Spinner;
