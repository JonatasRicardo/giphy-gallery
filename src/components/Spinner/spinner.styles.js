import {
  createUseStyles
} from "react-jss";

const useStyles = createUseStyles({
  spinner: {
    width: '40px',
    height: '40px',
    position: 'relative'
  },
  doubleBounce: {
    width: '100%',
    height: '100%',
    borderRadius: '50%',
    backgroundColor: ({
      theme
    }) => theme.dark.primary,
    opacity: '0.6',
    position: 'absolute',
    top: '0',
    left: '0',
    animation: '$sk-bounce 2.0s infinite ease-in-out',
  },
  doubleBounce2: {
    animationDelay: '-1.0s',
  },
  '@keyframes sk-bounce': {
    '0%, 100%': {
      transform: 'scale(0.0)'
    },
    '50%': {
      transform: 'scale(1.0)'
    }
  }
});

export default useStyles;
