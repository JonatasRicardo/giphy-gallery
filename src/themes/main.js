const theme = {
  dark: {
    background: "#000",
    primary: "#fff"
  },
  light: {
    background: "#fff",
    primary: "#000"
  },
  spacing: 10
};

export default theme;
