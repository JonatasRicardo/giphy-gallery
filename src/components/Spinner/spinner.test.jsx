import React from 'react';
import { ThemeProvider } from "react-jss";
import {
  render
} from '@testing-library/react';
import Spinner from './index';
import themes from '../../themes/main';


const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <Spinner />
    </ThemeProvider>
  );

test('should match snapshot', async () => {
  const { asFragment } = renderComponent({ theme: themes });
  expect(asFragment()).toMatchSnapshot()
});
