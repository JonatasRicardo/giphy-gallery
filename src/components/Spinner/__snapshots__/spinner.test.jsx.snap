// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner-0-2-1"
  >
    <div
      class="doubleBounce-0-2-2 doubleBounce-d0-0-2-5"
    />
    <div
      class="doubleBounce-0-2-2 doubleBounce-d0-0-2-5,doubleBounce2-0-2-3"
    />
  </div>
</DocumentFragment>
`;
